bin/parse-legacy-file

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Get version from .ruby-version file. .ruby-version is used by rbenv and now rvm.
+ruby_version_file=".ruby-version"
+get_ruby_version() {
+  local root="$1"
+  while [ -n "$root" ]; do
+    if [ -e "${root}/$ruby_version_file" ]; then
+      echo "${root}/$ruby_version_file"
+      exit
+    fi
+    root="${root%/*}"
```

Comment:
We shouldn't be traversing the file system in this script. The guide says

> The script will be passed the file path as its first argument.

We just need to take the file path argument, parse the file if necessary, and print the version. Since we have different parsing logic for different version file types we'll need to look at the filename in the argument to determine whether or not it's a Gemfile.

Reply:
See: https://github.com/asdf-vm/asdf/blob/master/docs/creating-plugins.md#binparse-legacy-file

Reply:
I was adapting what rbenv does. Happy to remove. The difference is what happens if you go to a subdirectory and try to execute ruby.

Reply:
Right, that is all handled by `asdf` itself. asdf will execute this file when necessary with the path to the Gemfile or .ruby-version file.

Reply:
Thanks for the feedback. I've updated.